kit/themes/android/src/main/java/tonlabs/uikit/themes/UIKitThemesAndroidNavigationBarModule.java

```diff
@@ -0,0 +1,99 @@
+package tonlabs.uikit.themes;
+
+import android.os.Build.VERSION;
+import android.os.Build.VERSION_CODES;
+
+import android.graphics.Color;
+import android.view.View;
+import android.view.Window;
+
+import androidx.annotation.NonNull;
+import androidx.core.view.ViewCompat;
+import androidx.core.view.WindowCompat;
+import androidx.core.view.WindowInsetsControllerCompat;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.module.annotations.ReactModule;
+
+import org.jetbrains.annotations.NotNull;
+
+@ReactModule(name = UIKitThemesAndroidNavigationBarModule.REACT_CLASS)
+public class UIKitThemesAndroidNavigationBarModule extends ReactContextBaseJavaModule {
+    protected static final String REACT_CLASS = "UIKitThemesAndroidNavigationBarModule";
+
+    UIKitThemesAndroidNavigationBarModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+    }
+
+    @NonNull
+    @NotNull
+    @Override
+    public String getName() {
+        return UIKitThemesAndroidNavigationBarModule.REACT_CLASS;
+    }
+
+    @ReactMethod
+    public void setAppearance(String theme) {
+        getReactApplicationContext().runOnUiQueueThread(new Runnable() {
+            @Override
+            public void run() {
+                Window window;
+
+                try {
+                    window = getCurrentActivity().getWindow();
+                } catch (NullPointerException err) {
+                    return;
+                }
+
+                View rootView = getCurrentActivity().getWindow().getDecorView().findViewById(android.R.id.content);
+                WindowInsetsControllerCompat insetsController = ViewCompat.getWindowInsetsController(rootView);
+
+                if (insetsController == null) {
+                    return;
+                }
+
+                insetsController.setAppearanceLightNavigationBars("dark-content".equals(theme));
+
+                if (VERSION.SDK_INT >= VERSION_CODES.Q) {
+                    window.setNavigationBarColor(Color.TRANSPARENT);
+                    /**
+                     * On Xiaomi there is a border on "dark-content", when a navbar is transparent
+                     * Trying to remove it here
+                     */
+                    if ("xiaomi".equalsIgnoreCase(android.os.Build.MANUFACTURER)) {
+                        window.setNavigationBarDividerColor(Color.parseColor("#01000000")); // Almost transparent
+                    }
+
+                    return;
+                }
+
```

Comment:
😅

Reply:
And at least on my phone it still doesn't work 😅Left in case it helps for some devices